src/database/database.cpp

```diff
@@ -73,7 +138,22 @@ void DatabaseManager::open_database() {
 void DatabaseManager::create_tables_from_schema() {
     std::string schema = read_schema_from_file(schema_file_path_);
     char* err_msg;
-    int return_status = sqlite3_exec(db_, schema.c_str(), nullptr, nullptr, &err_msg);
+    std::cout << "Creating tables..." << std::endl;
+    std::cout << schema.c_str() << std::endl;
+    std::string sch = "CREATE TABLE Loop ("
+                "loop_id INTEGER PRIMARY KEY,"
+                "loop_name TEXT UNIQUE,"
+                "loop_length TEXT,"
+                "loop_bpm INTEGER);"
+
+            "CREATE TABLE Track ("
+                "track_id INTEGER PRIMARY KEY,"
+                "loop_id INTEGER,"
+                "track_name TEXT UNIQUE,"
+                "track_type TEXT,"
+                "track_file_path TEXT,"
+                "FOREIGN KEY (loop_id) REFERENCES Loop(loop_id))";
+    int return_status = sqlite3_exec(db_, sch.c_str(), nullptr, nullptr, &err_msg);

```

Comment:
The unit tests that are written test if we can load the database from schema. If the database changes, it will save it to a db file and load from that. I am not sure what troubles you were having, but we can talk about it further to attempt to diagnose it